app/Models/Foundation/Summit/Events/Presentations/Presentation.php

```diff
@@ -339,31 +334,84 @@ public function getClassName()
      */
     public function getSpeakers()
     {
-        return $this->speakers;
+        return $this->speakers->map(function ($entity) {
+            return $entity->getSpeaker();
+        });
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getSpeakerAssignmentsMaxOrder(): int
+    {
+        $criteria = Criteria::create();
+        $criteria->orderBy(['order' => 'DESC']);
+        $res = $this->speakers->matching($criteria)->first();
+        return $res === false ? 0 : $res->getOrder();
     }
 
     /**
      * @param PresentationSpeaker $speaker
      */
     public function addSpeaker(PresentationSpeaker $speaker)
     {
-        if ($this->speakers->contains($speaker)) return;
-        $this->speakers->add($speaker);
-        $speaker->addPresentation($this);
+        $speakers = $this->getSpeakers();
+        if ($speakers->contains($speaker)) return;
```

Comment:
this comparison is not correct
u need to perform a query criteria

Reply:
done